django/thunderstore/social/providers.py

```diff
@@ -0,0 +1,112 @@
+from typing import Dict
+
+import requests
+from django.conf import settings
+
+
+class BaseOauthHelper:
```

Comment:
I'm not 100% sure what the "proper" way to declare abstract classes is nowadays, but at least https://docs.python.org/3/library/abc.html exists & you might want to check how mypy / type hints can be done appropriately for abstract methods that _need_ to be overwritten by subclasses. AFAIK that should be supported by mypy.

Reply:
Now uses ABC.